crowbar_framework/app/models/service_object.rb

```diff
@@ -970,6 +975,20 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     new_elements = new_deployment["elements"]
     element_order = new_deployment["element_order"]
 
+    # Build a list of old elements.
+    # elements_expanded on the old role is guaranteed to exists, as we already
+    # ran through apply_role with the old_role.  Cache is used for the case
+    # when pacemaker barclamp is deactivated.  elements_expanded gets updated
+    # by pacemaker barclamp.
+    old_elements = {}
```

Comment:
Im not 100% sure that moving this up here does not have any repercussions. I checked the code and I dont think it has, especially because I left the `element_order = old_deployment["element_order"] if (!old_deployment.nil? and element_order.nil?)` line in place, which seems the one that needs to be there due to the comment found on line 1034-1035:

```
 # save list of expanded elements, as this is needed when we look at the old
     # role. See below the comments for old_elements.
# role. See below the comments for old_elements.
```


@vuntz is this ok-ish at 100% to be moved?

Reply:
Looks safe.